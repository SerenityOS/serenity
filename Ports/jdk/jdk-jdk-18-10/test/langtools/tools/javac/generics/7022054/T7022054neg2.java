/*
 * @test /nodynamiccopyright/
 * @bug 7022054
 *
 * @summary  Invalid compiler error on covariant overriding methods with the same erasure
 * @compile/fail/ref=T7022054neg2.out -XDrawDiagnostics T7022054neg2.java
 *
 */

class T7022054neg2 {
    static class A {
        static A m(String s) { return null; }
    }
    static class B extends A {
        static <X extends String> A m(X s) { return null; }
    }
}
