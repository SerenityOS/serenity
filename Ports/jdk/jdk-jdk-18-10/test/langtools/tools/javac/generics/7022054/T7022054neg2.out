T7022054neg2.java:15:37: compiler.err.name.clash.same.erasure.no.hide: <X>m(X), T7022054neg2.B, m(java.lang.String), T7022054neg2.A
1 error
